### PR TITLE
docs: Deployment guide for CLI

### PR DESCRIPTION
**Pull request by wardbell**
See Issue # 19510
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: aio/content/guide/deployment.md [hunk 1]
```diff
@@ -1,319 +1,127 @@
...
-(3) Add `systemjs.config.server.js` (shown in the code sample below) to the `src/` folder.
-This alternative version configures _SystemJS_ to load _UMD_ versions of Angular
-(and other third-party packages) from the web.
-
-Modify `systemjs.config.server.js` as necessary to stay in sync with changes
-you make to `systemjs.config.js`.
-
-Notice the `paths` key:
-
+For the simplest deployment, build for development and xcopy the output directory to a web server.
```
Comment (filipesilva) [inline aio/content/guide/deployment.md]: `xcopy` -> `copy`

File: aio/content/guide/deployment.md [hunk 2]
```diff
@@ -1,319 +1,127 @@
...
-which describes running the AOT compiler from the command line
-and using [_rollup_](guide/deployment#rollup) for bundling, minification, uglification and tree shaking.
-
-
-{@a webpack}
-
+* Dead code elimination: removes unreferenced modules and much unused code.
 
-### Webpack (and AOT)
+The remaining [xcopy deployment steps](#xcopy) are the same as before.
```
Comment (filipesilva) [inline aio/content/guide/deployment.md]: `xcopy`->`copy`

File: aio/content/guide/deployment.md [hunk 3]
```diff
@@ -1,319 +1,127 @@
...
-If a library exports something that the application doesn't import, a tree shaking tool removes it from the code base.
+Switching to _production mode_ can make it run faster by disabling development specific checks such as the dual change detection cycles.
 
-Tree shaking was popularized by
-<a href="http://rollupjs.org/" title="Rollup">Rollup</a>, a popular tool with an ecosystem of
-plugins for bundling, minification, and uglification.
-Learn more about tree shaking and dead code elmination in
-<a href="https://medium.com/@Rich_Harris/tree-shaking-versus-dead-code-elimination-d3765df85c80#.15ih9cyvl" title="Tree-shaking and Dead Code Elimination">
-this post</a> by rollup-creator, Rich Harris.
+Building for production (or appending the `--environment=prod` flag) enables _production mode_
```
Comment (filipesilva) [inline aio/content/guide/deployment.md]: `--environment=prod` -> `--target=prod`

These two are actually different: https://github.com/angular/angular-cli/wiki/build#build-targets-and-environment-files

File: aio/content/guide/deployment.md [hunk 4]
```diff
@@ -328,119 +136,102 @@ Chrome DevTools Network Performance page</a> is a good place to start learning a
...
-If you do that, the module will be loaded immediately.
+To relocate the output, reconfigure the CLI by setting a different `outDir` in `.angular-cli.json`.
+</div>
 
-The bundling configuration must take lazy loading into consideration.
-Because lazy loaded modules aren't imported in JavaScript (as just noted), bundlers exclude them by default.
-Bundlers don't know about the router configuration and won't create separate bundles for lazy loaded modules.
-You have to create these bundles manually.
+The CLI **`ng serve`** command builds, watches, and serves the application from a local development server. 
+Unlike `build`, **it erases the output directory** and serves the build artifacts from memory instead of from files for a faster development experience.
```
Comment (filipesilva) [inline aio/content/guide/deployment.md]: `ng build` also deletes the directory, it just outputs new files there after.

File: aio/content/guide/deployment.md [hunk 5]
```diff
@@ -328,119 +136,102 @@ Chrome DevTools Network Performance page</a> is a good place to start learning a
...
+**After `ng serve`, the `dist/` folder won't have any files to xcopy.**
 
-The
-[Angular Ahead-of-Time Webpack Plugin](https://github.com/angular/angular-cli/tree/master/packages/%40ngtools/webpack)
-automatically recognizes lazy loaded `NgModules` and creates separate bundles for them.
+If you want to preserve the previously generated files, 
+append the `-no-delete-output-path` flag to the `serve` command.
+Note that the `ng serve` does NOT update generated files as you make changes.
```
Comment (filipesilva) [inline aio/content/guide/deployment.md]: You can use `ng build --watch` to continuously generate files.

**Review by filipesilva — CHANGES_REQUESTED** [review]

File: aio/content/guide/deployment.md (hunk 3 above)
Comment (wardbell) [reply aio/content/guide/deployment.md]: I just compared `--environment=prod` vs `--environment=dev` and they enable/disable `prodMod` exactly as I said and expected.

So do `--target=production` vs `--target=development`. But the `--environment` flag is way more intuitive, especially after looking at `main.ts`
```
if (environment.production) {
  enableProdMode();
}
```
Why should I change from `--environment` to `--target`?  The wiki page is not helping me understand.

Comment (filipesilva) [reply aio/content/guide/deployment.md]: You're right, I misunderstood what was being talked about here.

File: aio/content/guide/deployment.md [hunk 6]
```diff
@@ -482,75 +282,65 @@ The list is by no means exhaustive, but should provide you with a good starting
 * [Lite-Server](https://github.com/johnpapa/lite-server): the default dev server installed with the
 [Quickstart repo](https://github.com/angular/quickstart) is pre-configured to fallback to `index.html`.
```
Comment (Foxandxss) [inline aio/content/guide/deployment.md]: We will deprecate the quickstart, we shouldn't name it anymore.

**Review by Foxandxss — APPROVED** [review]

**Review by filipesilva — APPROVED** [review]

**Comment by vicb** [comment]
missing:
- target label
- merge label
- green CI

**Comment by IgorMinar** [comment]
@vicb the ci failure is a flake and unrelated to this change.

**Comment by IgorMinar** [comment]
I fixed the labels. @wardbell please always apply "target" and "merge" label when your pr is ready for merge.

**Comment by Foxandxss** [comment]
Yes sorry, I have too many PRs and travis is being flaky all day so I go back and forward restarting them and checking all the statuses.

**Pull request merged**